idaes/models/flowsheets/tests/test_methanol_flowsheet.py

```diff
@@ -0,0 +1,673 @@
+    assert hasattr(model.fs, "operating_cost")
+    assert hasattr(model.fs.R101, "L_eq")
+    assert hasattr(model.fs.H101, "cost_heater")
+    assert hasattr(model.fs.H102, "cost_heater")
+    assert hasattr(model.fs, "annualized_capital_cost")
+    assert hasattr(model.fs, "sales")
+    assert hasattr(model.fs, "raw_mat_cost")
+    assert hasattr(model.fs, "objective")
+    assert degrees_of_freedom(model) == 0
+
+
+@pytest.mark.component
+def test_optimize_with_costing(model):
+
+    # Set up Optimization Problem (Maximize Revenue)
+    # keep process pre-reaction fixed and unfix some post-process specs
+    model.fs.R101.conversion.unfix()
+    model.fs.R101.conversion_lb = Constraint(
+        expr=model.fs.R101.conversion >= 0.75)
+    model.fs.R101.conversion_ub = Constraint(
+        expr=model.fs.R101.conversion <= 0.85)
+    model.fs.R101.outlet_temp.deactivate()
+    model.fs.R101.outlet_t_lb = Constraint(
+        expr=model.fs.R101.control_volume.properties_out[0.0].temperature
+        >= 405)
+    model.fs.R101.outlet_t_ub = Constraint(
+        expr=model.fs.R101.control_volume.properties_out[0.0].temperature
+        <= 505)
+
+    # Optimize turbine work (or delta P)
+    model.fs.T101.deltaP.unfix()  # optimize turbine work/pressure drop
+    model.fs.T101.outlet_p_lb = Constraint(
+        expr=model.fs.T101.outlet.pressure[0] >= 10E5)
+    model.fs.T101.outlet_p_ub = Constraint(
+        expr=model.fs.T101.outlet.pressure[0] <= 51E5*0.8)
+
+    # Optimize Cooler outlet temperature - unfix cooler outlet temperature
+    model.fs.H102.outlet_temp.deactivate()
+    model.fs.H102.outlet_t_lb = Constraint(
+        expr=model.fs.H102.control_volume.properties_out[0.0].temperature
+        >= 407.15*0.8)
+    model.fs.H102.outlet_t_ub = Constraint(
+        expr=model.fs.H102.control_volume.properties_out[0.0].temperature
+        <= 480)
+
+    model.fs.F101.deltaP.unfix()  # allow pressure change in streams
+
+    model.fs.F101.isothermal = Constraint(
+        expr=model.fs.F101.control_volume.properties_out[0].temperature ==
+        model.fs.F101.control_volume.properties_in[0].temperature)
+
+    solver = get_solver()
+    optarg = {'tol': 1e-6,
+              'max_iter': 5000}
+    solver.options = optarg
+    results2 = solver.solve(model, tee=True)
+    assert results2.solver.termination_condition == \
+        TerminationCondition.optimal
+
+    assert value(model.fs.R101.rate_reaction_extent[0, "R1"]) == pytest.approx(
+        269.2805, abs=1e-2)
+    assert value(model.fs.R101.heat_duty[0])/1E6 == pytest.approx(
+        -51.3636, abs=1e-2)
+    assert value(model.fs.T101.work_isentropic[0])/1E6 == pytest.approx(
+        -1.9905, abs=1e-2)
+    assert value(model.fs.F101.recovery*100) == pytest.approx(
+        95.5433, abs=1e-2)
+    assert value(model.fs.objective)/1E6 == pytest.approx(
+        81.0476, abs=1e-2)
+
+
+@pytest.mark.unit
+def test_report(model):
+    stream = StringIO()
```

Comment:
I know I said to do this last time, but we have come to realize that this can be a bit too fragile. However, with the way the `report` method is currently written I don't think there are many alternatives.